lib/timezone/lookup/geonames.rb

```diff
@@ -34,13 +34,37 @@ def lookup(lat, long)
         raise(Timezone::Error::GeoNames, e.message)
       end
 
+      def lookup_with_etc(lat, long)
+        response = client.get(url(lat, long))
+
+        return unless response.body
+
+        data = JSON.parse(response.body)
+
+        return data['timezoneId'] if data['timezoneId']
+
+        return get_etc(data['gmtOffset']) if data['gmtOffset']
+
+        return unless data['status']
+
+        raise(Timezone::Error::GeoNames, data['status']['message'])
+      rescue => e
+        raise(Timezone::Error::GeoNames, e.message)
+      end
+
       private
 
+      def get_etc(offset)
+        return 'Etc/UTC' if offset == 0
+        "Etc/GMT#{format('%+d', offset.to_i)}"
```

Comment:
This seems like good default behavior. I don't see why it only needs to be implemented with Geonames?

Even if it is only implemented with geonames, why should this not be included as part of the default `lookup` function?